Msi dts profiles

miczyg1 opened the pull request.

miczyg1 committed:
dts/profiles: Update profiles for MSI releases 0.9.4 and 1.1.6

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

m-iwanicki commented on dts/profiles/msi-pro-z690-a-ddr5 UEFI Update - DPP.profile:
```diff
@@ -4,7 +4,6 @@ dmidecode -s baseboard-product-name 0
 dmidecode -s processor-version 0
 dmidecode -s bios-vendor 0
 dmidecode -s bios-version 0
-fsread_tool test -f /sys/firmware/efi/efivars/FirmwareUpdateMode-d15b327e-ff2d-4fc1-abf6-c12bd08c1359 1
```
I see you've tested this on newest non-rc release, this part was added later: https://github.com/Dasharo/dts-scripts/commit/ba25f2944992f83750dd39a9240dae4911e09c0a

I'm pretty sure [v2.7.2-rc1](https://github.com/Dasharo/meta-dts/releases/tag/v2.7.2-rc1) includes those changes.
Those tests/profiles are (or at least should be) synced with newest develop branch on `meta-dts`

DaniilKl replied:
ddbb29ecc70c1da93ed906eddb20ebc3c2e03bc4

Commit:
dts: profile: msi-pro-z[67]90*ddr5 UEFI Update - DPP: get FUM check back

The FUM check is not present in DTS v2.7.1 on wich the profiles were
originally collected. The check was added in commit
ba25f2944992f83750dd39a9240dae4911e09c0a in Dasharo/dts-scripts repo.
and will be present in the next stable DTS version.
This commit adds the FUM check back into profile to be compatible with
the DTS on develop branch in Dasharo/meta-dts.

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl commented:
The profile `msi-pro-z690-a-wifi-ddr4 UEFI Update - DPP.profile` has not been updated because it was not possible to collect it due to the platform inaccessibility.

Commit:
dts: profile: msi-pro-z690-a-wifi-ddr4 UEFI Update - DPP.profile: make
it outdated

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

m-iwanicki commented on dts/profiles/msi-pro-z790-p-ddr5 UEFI Update - DPP.profile:
```diff
@@ -22,25 +22,6 @@ dmidecode -s bios-version 0
-futility show /tmp/bios.bin 0
-flashrom -p internal --ifd -i bios -i fd -i me -w /tmp/biosupdate 0
-flashrom -p internal --ifd -i bios -i fd -i me -w /tmp/biosupdate 0
+cap_upd_tool /tmp/biosupdate 0
```
@DaniilKl I tested on QEMU and DTS `2.7.2-rc1` and I got normal update, not capsule one

m-iwanicki replied:
Same with meta-dts CI https://github.com/Dasharo/meta-dts/actions/runs/20059361209/job/57561578326:

`E2E057: msi-pro-z690-a-ddr5 UEFI Update - DPP`:

```
Following firmware will be used to deploy Dasharo:
Dasharo BIOS firmware:
  - link: dasharo-msi-uefi/MS-7D25/v1.1.6/msi_ms7d25_v1.1.6_ddr5.rom
  - hash: 72f438f99cc6e2b205f3c9424c21a02cf53abc844b42f6db3661fee5f6468ffd

You can learn more about this release on: https://docs.dasharo.com/

Do you want to deploy this Dasharo Firmware on your platform [y|n]
```

<img width="1399" height="180" alt="image" src="https://github.com/user-attachments/assets/10300d67-3209-4f42-9ea7-6adb244f2aff" />

Needs to be debugged before merging.

DaniilKl replied:
As discussed internally, the problem is with the `DTS_TEST_VERSIONS` that should contain version from which the capsules are supported for these platforms.

m-iwanicki replied:
52ea41c460fd

```
==============================================================================
Dts-E2E
==============================================================================
Create tests                                                          | PASS |
------------------------------------------------------------------------------
[ WARN ] Workflow isn't configured for profile verification.
E2E001: msi-pro-z690-a-ddr5 UEFI Update - DCR                         | PASS |
------------------------------------------------------------------------------
E2E002: msi-pro-z690-a-ddr5 UEFI Update - DPP                         | PASS |
------------------------------------------------------------------------------
E2E003: msi-pro-z690-a-wifi-ddr4 UEFI Update - DCR                    | PASS |
------------------------------------------------------------------------------
E2E004: msi-pro-z690-a-wifi-ddr4 UEFI Update - DPP                    | FAIL |
Teardown failed:
Profiles are not identical!: 1 != 0
------------------------------------------------------------------------------
E2E005: msi-pro-z790-p-ddr5 UEFI Update - DPP                         | PASS |
------------------------------------------------------------------------------
Dts-E2E                                                               | FAIL |
6 tests, 5 passed, 1 failed
==============================================================================
```

And in `log.html`:

```
Following firmware will be used to deploy Dasharo:
Dasharo BIOS firmware:
  - link: dasharo-msi-uefi/MS-7E06/v0.9.4/msi_ms7e06_v0.9.4_ddr5.cap
  - hash: 587f1df8b4f0794dbae91172b6ce0d06669a40e4901fae8c5b855948dd783b46

You can learn more about this release on: https://docs.dasharo.com/

Do you want to deploy this Dasharo Firmware on your platform [y|n]
```

Can you verify?

DaniilKl replied:
Thank you! Results:

```bash
==============================================================================
Dts-E2E                                                                       
==============================================================================
Create tests                                                          | PASS |
------------------------------------------------------------------------------
E2E001: msi-pro-z690-a-wifi-ddr4 UEFI Update - DPP                    | FAIL |
Teardown failed:
Profiles are not identical!: 1 != 0
------------------------------------------------------------------------------
E2E002: msi-pro-z690-a-ddr5 UEFI Update - DPP                         | PASS |
------------------------------------------------------------------------------
E2E003: msi-pro-z790-p-ddr5 UEFI Update - DPP                         | PASS |
------------------------------------------------------------------------------
Dts-E2E                                                               | FAIL |
4 tests, 3 passed, 1 failed
==============================================================================
```

The E2E001 failed because of this: https://github.com/Dasharo/open-source-firmware-validation/pull/1166#issuecomment-3626438373 .

So I think this PR is LGMT.

P.S. I wonder whether we should do [the same](https://github.com/Dasharo/open-source-firmware-validation/pull/1203) for `E2E001: msi-pro-z690-a-wifi-ddr4 UEFI Update - DPP`, cos we do not have a platform to test on. I think, considering the capsule update, it is safe to do a "theoretical prove" of the profile.

ghost commented:
I ran the update on physical platform and can confirm the profile is the same and capsule update is default route.

--edit--

Same image defaults to flashrom on qemu.

m-iwanicki committed:
platform-configs: dts: bump MSI update version so it tests capsules

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>
